course_catalog/tasks.py

```diff
@@ -35,26 +36,49 @@ def get_edx_data(force_overwrite=False):
         settings.EDX_API_URL
         and settings.EDX_API_CLIENT_ID
         and settings.EDX_API_CLIENT_SECRET
+        and settings.OCW_LEARNING_COURSE_BUCKET_NAME
+        and settings.OCW_LEARNING_COURSE_ACCESS_KEY
+        and settings.OCW_LEARNING_COURSE_SECRET_ACCESS_KEY
     ):
-        log.warning("Required settings missing for get_edx_data")
+        log.warning("Required settings missing for sync_and_upload_edx_data")
         return
     url = settings.EDX_API_URL
     access_token = get_access_token()
+    should_upload_to_s3 = True
+
+    edx_data = {"count": 0, "catalog_url": settings.EDX_API_URL, "results": []}
 
     while url:
+        log.info("Syncing edX data...")
```

Comment:
Can you move this so it runs before the `while` loop, not inside of it?

Reply:
Definitely.